packages/snap-client/src/Client/transforms/searchResponse.ts

```diff
@@ -242,7 +242,7 @@ transformSearchResponse.result = (rawResult: rawResult): SearchResponseModelResu
 			core: coreFieldValues,
 		},
 		attributes,
-		badges: typeof rawResult.badges == 'object' && !Array.isArray(rawResult.badges) ? rawResult.badges : [],
+		badges: typeof rawResult.badges == 'object' && Array.isArray(rawResult.badges) && typeof rawResult.badges[0] == 'object' ? rawResult.badges : [],
```

Comment:
Seems like the logic for checking `typeof rawResult[k] == 'object'` is unnecessary, as the next check is if it is an array.